bootstrap/v2/pkg/apis/apps/kfdef/v1alpha1/application_types.go

```diff
@@ -47,6 +71,62 @@ var DefaultRegistry = RegistryConfig{
 	Path: "kubeflow",
 }
 
+// Plugin can be used to customize the generation and deployment of Kubeflow
+// TODO(jlewi): Should Plugin contain K8s TypeMeta so that we can use ApiVersion and Kind
+// to identify what it refers to?
+//
+// We disable deep-copy-gen because it chokes on type interface{}.
+// What are the implications of that? Will we eventually need to write our own DeepCopy
+// method based on marshling the object to bytes?
+//
+type Plugin struct {
+	Name       string            `json:"name,omitempty"`
+
+	// TODO(jlewi): Should we be using runtime.Object or runtime.RawExtension
+	Spec *runtime.RawExtension `json:"spec,omitempty"`
+}
+
+// SecretRef is a reference to a secret
+type SecretRef struct {
+	// Name of the secret
+	Name string `json:"name,omitempty"`
+}
+
+// Repo provides information about a repository providing config (e.g. kustomize packages,
+// Deployment manager configs, etc...)
+type Repo struct {
+	// Name is a name to identify the repository.
+	Name string `json:"name,omitempty"`
+	// URI where repository can be obtained.
+	// Can use any URI understood by go-getter:
+	// https://github.com/hashicorp/go-getter/blob/master/README.md#installation-and-usage
+	Uri string `json:"uri,omitempty"`
+
+	// Root is the relative path to use as the root.
```

Comment:
No. This refers to the source for the repo. So repo would point to 
https://github.com/kubeflow/kubeflow/archive/master.zip

And root might be "master/"
because GitHub creates a folder in the zip file with the branch name

Reply:
I see thanks